velox/vector/arrow/c/Bridge.h

```diff
@@ -18,7 +18,7 @@
 
 #include "velox/common/memory/Memory.h"
 #include "velox/vector/BaseVector.h"
-#include "velox/vector/arrow/Abi.h"
+#include "velox/vector/arrow/c/abi.h"
```

Comment:
Any particular reason to change the file name from Abi.h to abi.h?

Reply:
To be the same with arrow's abi header and avoid conflict. Because the two abi headers in Velox and Arrow have the same content but different pathes and names, redefinition will occur if including both. We changed them to be the same, so the compiler will just use the first found one.

Reply:
In Arrow, there is an abi header with the same content ([link](https://github.com/apache/arrow/blob/master/cpp/src/arrow/c/abi.h)). If including both, the definition of, for example, ArrowSchema, is duplicated. The compiler do not know which we want to refer to. Another solution is to change one of them from ArrowSchema to another name, but the solution in this PR may require less efforts.

Reply:
What I don't understand is how would "#pragma once" treat these 2 different files as the same? Do you know?

Reply:
The "#pragma once" can only control the compilation of Velox libraries, but our code also depends on Arrow's libraries, in which there is also an abi header. In this case, the definitions of the two headers are brought together.

Reply:
I'm still trying to understand. You are still going to include both files, but because the definitions will match (as long as the files are identical copies), the compilation will succeed? Feels fragile. Is there a more robust solution?

Reply:
Yes, because our code is like "#include arrow/c/abi.h", if we add velox/vector into the include path, compiler will just use the first found one. It may be not a good practice to have two different headers with the same definition, but when we use both Velox and Arrow, this situation happens. Another solution we can think of is to change one of them to be a different one.

Reply:
@rui-mo Rui, thank you for explaining. This trick is indeed not a best practice and feel quite fragile. 

> Another solution we can think of is to change one of them to be a different one.

Let's try this. What what would look like?

Reply:
That is to use different names for these definitions, for example,
from ArrowSchema to VeloxArrowSchema.

Reply:
A third solution can be we add arrow as build-time dependency, but only use the C ABI header and nothing else, so we don't depends on it in runtime.

Reply:
Looks like the ARROW_C_DATA_INTERFACE macro guard was recently introduced to arrow (after we made our drop-in copy of the C ABI file). Adding these two new lines to the Abi.h file should solve the issue. 

For context, check: https://github.com/apache/arrow/pull/13115

and: https://github.com/apache/arrow/blob/master/docs/source/format/CDataInterface.rst

"Note: The canonical guard ARROW_C_DATA_INTERFACE is meant to avoid duplicate definitions if two projects copy the C data interface definitions in their own headers, and a third-party project includes from these two projects. It is therefore important that this guard is kept exactly as-is when these definitions are copied."

@rui-mo could you update this PR to just add the two macros?

Reply:
@pedroerp These 2 macros are not included in the latest release version (8.0.0) yet